File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/paper/dto/PaperAiSaveDto.java
package com.yyjy.exam.entity.paper.dto;

import com.yyjy.exam.entity.question.entity.QuestionType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PaperAiSaveDto {
	private String name;
	private String description;
	private int duration;
	private List<Rule> rules;
	
	@Data
	@NoArgsConstructor
	@AllArgsConstructor
	public static class Rule {
		private QuestionType type;
		private List<Long> categoryIds;
		private int count;
		private int score;
	}
}

File: exam-system-server/src/main/kotlin/com/yyjy/models/dto/PaperAiSaveDto.kt
package com.yyjy.models.dto

data class PaperAiSaveDto(
    val name: String,
    val description: String,
    val duration: Int,
    val rules: List<PaperAiSaveDtoRule>
)

data class PaperAiSaveDtoRule(
    val type: String,
    val categoryIds: List<Long>,
    val count: Int,
    val score: Int
)